test/linalg/rowvector.jl

```diff
@@ -0,0 +1,245 @@
+    @test (rv2 = copy(rv); rv2[1,2] = 6; rv2[2] === 6)
+
+    @test length(rv) === 3
+    @test size(rv) === (1,3)
+    @test size(rv,1) === 1
+    @test size(rv,2) === 3
+
+    @test map(-, rv)::RowVector == [-1 -2 -3]
+    @test (-).(rv)::RowVector == [-1 -2 -3]
+    @test (-).(rv,1)::RowVector == [0  1  2]
+
+    y = rand(Complex{Float64},3)
+    @test sum(abs2, imag.(diag(y .+ y'))) < 1e-20
+end
+
+@testset "Diagonal ambiguity methods" begin
+    d = Diagonal([1,2,3])
+    v = [2,3,4]
+    rv = v.'
+
+    @test (rv*d)::RowVector == [2,6,12].'
+    @test_throws Exception d*rv
+
+    @test (d*rv.')::Vector == [2,6,12]
+
+    @test_throws Exception rv.'*d
+
+    @test (d*rv')::Vector == [2,6,12]
+
+    @test_throws Exception rv'*d
+
+    @test (rv/d)::RowVector ≈ [2/1  3/2  4/3]
+
+    @test_throws Exception d \ rv
+
+end
+
+@testset "Bidiagonal ambiguity methods" begin
+    bd = Bidiagonal([1,2,3], [0,0], true)
+    v = [2,3,4]
+    rv = v.'
+
+    @test (rv/bd)::RowVector ≈ [2/1  3/2  4/3]
+
+    @test_throws Exception bd \ rv
+end
+
+@testset "hcat" begin
+    @test isa([([1, 2, 3].') 4], RowVector{Int})
+    @test isa([([1, 2, 3].') ([4, 5].')], RowVector{Int})
+end
+
+@testset "Left Division" begin
+    mat = diagm([1,2,3])
+    v = [2,3,4]
+    rv = v.'
+
+    @test_throws Exception mat \ rv
+end
+
+@testset "Multiplication" begin
+    v = [1,2,3]
+    rv = v.'
+    mat = diagm([1,2,3])
+
+    @test (rv*v) === 14
+    @test (rv*mat)::RowVector == [1 4 9]
+    @test_throws Exception [1]*reshape([1],(1,1)) # no longer permitted
+    @test_throws Exception rv*rv
+    @test (v*rv)::Matrix == [1 2 3; 2 4 6; 3 6 9]
+    @test_throws Exception v*v # Was previously a missing method error, now an error message
+    @test_throws Exception mat*rv
+
+    @test_throws Exception rv*v.'
+    @test (rv*mat.')::RowVector == [1 4 9]
+    @test_throws Exception [1]*reshape([1],(1,1)).' # no longer permitted
+    @test rv*rv.' === 14
+    @test_throws Exception v*rv.'
+    @test (v*v.')::Matrix == [1 2 3; 2 4 6; 3 6 9]
+    @test (mat*rv.')::Vector == [1,4,9]
+
+    @test (rv.'*v.')::Matrix == [1 2 3; 2 4 6; 3 6 9]
+    @test_throws Exception rv.'*mat.'
+    @test (v.'*mat.')::RowVector == [1 4 9]
+    @test_throws Exception rv.'*rv.'
+    @test v.'*rv.' === 14
+    @test_throws Exception v.'*v.'
+    @test (mat.'*rv.')::Vector == [1,4,9]
+
+    @test_throws Exception rv.'*v
+    @test_throws Exception rv.'*mat
+    @test (v.'*mat)::RowVector == [1 4 9]
+    @test (rv.'*rv)::Matrix == [1 2 3; 2 4 6; 3 6 9]
+    @test_throws Exception v.'*rv
+    @test v.'*v === 14
+    @test_throws Exception mat.'*rv
+
+    z = [1+im,2,3]
+    cz = z'
+    mat = diagm([1+im,2,3])
+
+    @test cz*z === 15 + 0im
+
+    @test_throws Exception cz*z'
+    @test (cz*mat')::RowVector == [-2im 4 9]
+    @test_throws Exception [1]*reshape([1],(1,1))' # no longer permitted
+    @test cz*cz' === 15 + 0im
+    @test_throws Exception z*vz'
```

Comment:
for example, `vz` is a typo and this test still passed because `Exception` is far too broad a type to be testing for